generate_figures/utilities.py

```diff
@@ -0,0 +1,55 @@
+import pandas as pd
+import wandb
+
+def plot_params(figsize, fontsize, ticksize):
+    params = {
+        "text.usetex": True,
+        "font.family": "serif",
+        "figure.figsize": figsize,
+        "figure.dpi": 80,
+        "figure.edgecolor": "k",
+        "font.size": fontsize,
+        "axes.labelsize": fontsize,
+        "axes.titlesize": fontsize,
+        "xtick.labelsize": ticksize,
+        "ytick.labelsize": ticksize,
+    }
+    return(params)
+
+
+
+def get_results_by_tag(project, tag, test_results = False):
+    api = wandb.Api()
+    count=0
+    runs = api.runs(project, filters={"tags": tag})
+    df_test_results = pd.DataFrame()
+    for i in range(len(runs)):
+        count+=1
+        df = pd.DataFrame()
+        cols = dict(runs[i].summary)
+        cols.update(dict(runs[i].config))
+        cols['id'] = runs[i].id
+        df = pd.DataFrame.from_dict(cols, orient='index').T
```

Comment:
Wny create the `df` here?  Shouldn't you create the dataframe only you have (optionally) got the test_results?